pandas/util/_exceptions.py

```diff
@@ -11,6 +11,10 @@
     from collections.abc import Generator
     from types import FrameType
 
+Pandas4DeprecationWarning = DeprecationWarning
+Pandas5DeprecationWarning = DeprecationWarning
+CurrentDeprecationWarning = Pandas4DeprecationWarning
```

Comment:
I think this should be implemented as a `Generic` class with a parameter corresponding to the version number.  So you'd have `PandasDeprecationWarning[4]` and `PandasDeprecationWarning[5]`, etc.  This would allow people to write code that would catch a `PandasDeprecationWarning` without having to worry about the particular version number.

Reply:
How will the class decide whether to inherit from `FutureWarning` or `DeprecationWarning`?